models/collection.js

```diff
@@ -0,0 +1,113 @@
+'use strict';
+
+const Joi = require('joi');
+const mutate = require('../lib/mutate');
+
+const MODEL = {
+    id: Joi
+        .number()
+        .integer()
+        .positive(),
+
+    userId: Joi
+        .number()
+        .integer()
+        .positive()
+        .description('User ID'),
+
+    name: Joi
+        .string()
+        .max(128)
+        .description('Collection name')
+        .example('Favorites'),
+
+    description: Joi
+        .string()
+        .max(256)
+        .description('Collection description')
+        .example('List of my favorite collections'),
+
+    pipelineIds: Joi
+        .array()
+        .items(Joi.number().integer().positive())
+};
+
+module.exports = {
+    /**
+     * All the available properties of Collection
+     *
+     * @property base
+     * @type {Joi}
+     */
+    base: Joi.object(MODEL).label('Collection'),
+
+    /**
+     * Properties for collection that will come back during a GET request
+     *
+     * @property get
+     * @type {Joi}
+     */
+    get: Joi.object(mutate(MODEL, [
+        'id',
+        'name',
+        'pipelineIds'
+    ], [
+        'description'
+    ])).label('Get collecion'),
+
+    /**
+     * Properties for Collection that will be passed during a CREATE request
+     *
+     * @property create
+     * @type {Joi}
+     */
+    create: Joi.object(mutate(MODEL, [
```

Comment:
I was initially thinking that creating a pipeline would just have like a name and an optional description and then pipelines could be added through update requests. But I can see how in an API standpoint it would be nice to support that functionality. I can see it being useful if we are going to go with the fork model for modifying someone else's collection.